### DISCUSSION TITLE
I want to make own APIs with Mozilla hubs for enterprise.

### DISCUSSION THREAD

**Discussion by jeffrey-kor** [mozilla/hubs#4540] [Hubs Cloud Questions] [answered]
Hello. everyone.

I would like to make own Apis with mozilla hubs for enterprise.

And, I want to add apis more 

My issue is what I don't know how do I run my local reticulum server locally.

How do I set up the test environment of mozilla hubs client and backend on my local?

[Link](https://github.com/mozilla/hubs/discussions/3323)

**Comment by pattersonbl2** [comment, marked as answer]
we don't support setting up Reticulum locally due to restrictions on developer credentials. Although relatively difficult and new territory, you're welcome to set up this up yourself. In addition to running reticulum, you'll need to also run [Hubs](https://github.com/mozilla/hubs) and [Dialog](https://github.com/mozilla/dialog) locally because the developer Dialog server is locked down and your local Reticulum will not connect properly. 

Reference [this discussion thread](https://github.com/mozilla/hubs/discussions/3323) for more information.
